pkg/addons/metadata/enable.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2020 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metadata
+
+import (
+	"os/exec"
+	"path/filepath"
+	"strconv"
+
+	"github.com/pkg/errors"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/minikube/pkg/minikube/assets"
+	"k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/driver"
+	"k8s.io/minikube/pkg/minikube/machine"
+)
+
+// EnableOrDisable enables or disables the metadata addon depending on the val parameter
+func EnableOrDisable(cfg *config.ClusterConfig, name string, val string) error {
+	enable, err := strconv.ParseBool(val)
+	if err != nil {
+		return errors.Wrapf(err, "parsing bool: %s", name)
+	}
+	if enable {
+		return enableAddon(cfg)
+	}
+	return disableAddon(cfg)
+
+}
+
+func enableAddon(cfg *config.ClusterConfig) error {
+	// This is the default location for GCP credentials to live, it's where they're stored when gcloud login is run
+	credsPath := filepath.Join(homedir.HomeDir(), ".config", "gcloud", "application_default_credentials.json")
```

Comment:
I found this function which seems more portable: https://godoc.org/golang.org/x/oauth2/google#FindDefaultCredentials

Reply:
I don't actually want the credentials, I just want the json file.

Reply:
Fixed.